backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java

```diff
@@ -60,27 +58,21 @@ public CassandraTableManager ensureAllTables() {
 
     public void clearAllTables() {
         CassandraAsyncExecutor executor = new CassandraAsyncExecutor(session);
-        FluentFutureStream.of(
-            module.moduleTables()
-                .stream()
+        Flux.fromIterable(module.moduleTables())
+                .publishOn(Schedulers.elastic())
                 .map(CassandraTable::getName)
-                .map(name -> truncate(executor, name)))
-            .join();
+                .flatMap(name -> truncate(executor, name))
+                .blockLast();
```

Comment:
we don't rely want the last value, we want the Flux to complete.
You can use `then().block()` instead, it closer to you intention